### DISCUSSION TITLE
Recover (Delete) Account For Org Owner Impossible If Authenticator Lost

### DISCUSSION THREAD

**Discussion by jklaus** [dani-garcia/vaultwarden#2579] [Q&A]
While I can still log into my vault via the master password, I am unable to log into the web vault as it requires authenticator and my phone went for a swim this past weekend.  Since I cannot log into the web vault I am unable to change Org Ownership, and since I cannot change Org Ownership I cannot recover/delete my account.  Please advise.

**Comment by BlackDex** [comment]
Both the options you mention are available via the admin interface, see: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page

Also, i suggest to try and enabled multiple MFA options if possible, so mail and authenticator for example.

**Reply by jklaus** [reply]
@BlackDex - I am not hosting my own, my org is hosted on their hardware.  I cannot log into my account on their web app as its wanting a code I cannot provide since my phone is bricked.. and I cannot recover/delete my account as I am the sole owner of an org which I cannot alter as I cannot log in.  It seems there is a bug in their overall logic in terms of account recovery/deletion & account ownership.

**Comment by BlackDex** [comment]
Then ask the maintainer to remove your MFA via the Admin Interface for your account. You still need the correct password though, so it will be less safe, but you should be able to login and not have it wide open for the world.

That is your only option.

**Reply by jklaus** [reply]
They have proposed deleting the org from my acct.  I responded back to them with a couple other options, including yours, so I'm waiting to hear back.  All that said, there is a logical issue with sole org owners and MFA in regards to account recovery/deletion.

**Comment by BlackDex** [comment]
Well, that is why there is an option to delete the MFA.
Also, you just can't remove an account if that is the sole owner of an organization, because you can't add a new person to an existing org with anybody in it.

I don't think there is a logic issue, since anything else would make it less secure. Again, that is where the MFA removal is for which a server admin could fix.

**Reply by jklaus** [reply]
To me, if a sole owner is trying to delete their account it should prompt and allow them to accept deleting the org(s) along with their user.  This would prevent said org owner from having to reach out to their support.  Ultimately that is what their support has to do anyway. (I just heard back, the only option is for them to delete the org and then me delete my account.)

**Reply by BlackDex** [reply]
You can delete an org if you are an owner and all other users lose access to it, since it doesn't exist anymore, but for that you need to be logged in. It would be rather stupid if you would get that option during a recovery procedure, that would make it possible for someone else to try and do that.

Regarding what the support is telling you, that is just not true, but if that is the only thing they want to do, that's not something Vaultwarden can fix for you.

Vaultwarden offers the option to remove MFA for these cases.

Also in the end it wouldn't matter to what you suggested or what the support team is doing now, in both cases you would lose the org and the data, so I'm not really seeing a win-win in your option.

I would go for better security and some manual adjustments instead of doing these actions fully automated. Don't get me wrong, i understand that for a sole user with no other users in the org it sounds a bit strange, but still, i wouldn't want that a default action.
For an org with just one owner and multiple other users, i wouldn't want to allow that at all anyways.

**Reply by jklaus** [reply]
You cannot delete the org if you are an owner and have MFA enabled but no longer have access to said MFA app, thus preventing you from being able to log into the web app.. or even logging into the CLI.  This is my point.

**Reply by BlackDex** [reply]
And that is where the delete MFA option is for in the Admin Interface.

**Reply by jklaus** [reply]
If you're hosting your own, yes.  For the SaaS side though, that solution isn't very good.  You have to reach out to their support to get things taken care of, and they won't disable MFA regardless.  Currently they're just manually removing the solely owned orgs.  The account recovery/deletion process should allow you to choose to delete your solely owned orgs as part of (technically ahead of) account deletion.  If they implemented that behavior then I would have been able to self service in my use-case.

**Reply by BlackDex** [reply]
Well, i disagree with that because of safety. Again, in the end it has the same outcome, and from my point of few i would still go with safety.

Vaultwarden is build for self hosted in mind, or maybe even small businesses. But not for SaaS. Also, even if Vaultwarden would build something like that, who doesn't say they remove that feature. I even don't know which version they are running, or what other adjustments they have made at all.

Lets just agree to disagree on this.
You are always free the create a PR with this feature and see it this gets merged, but I'm not for a functionality like this my self.

**Reply by jklaus** [reply]
What isn't safe/secure about allowing a user to delete their orgs when they're already trying to delete their user?

**Reply by BlackDex** [reply]
Well, Bitwarden (and Vaultwarden) just do not permit the deletion of a user which is a sole owner of an organization, a sole owner doesn't mean it's the only user in that org, but that person could be the currently only owner, and that just isn't allowed, period.
Also, what if you are the sole owner of multiple organizations, of which one is just only you as a user, and others have multiple users, then your account deletion would still not happen.

The reason is, is if have multiple organizations, and of which one is shared with other people who still need access where someone would forgot to hand over the ownership and you are still the only owner in that organization, and you want to leave, and delete your account, and Vaultwarden would just happily remove that shared organization, those other users can't access it anymore. That is the safety I'm talking about.

Also, for SaaS solutions, if you have account issues, you just need to call support, they should be there to help.

Anyway, in a normal procedure you would delete the organization first, and then delete your account, also because deletion of the organization via the web-vault requires that you provided your password again for safety checking.

But in your case, it's just a lost MFA, for which in the Admin Interface there is an option to delete all MFA's for that user!

Also, which I forgot to mention. You have been provided with a recovery token in case you would lose your MFA. That is something you should have stored somewhere safe and not on the same place as the rest of your secrets of course.

Again, i see your point of view regarding self-service. But again, Vaultwarden is build for self hosted solutions, not SaaS services in mind.  And i think in these cases contacting support is a right way to go for any service. And you may disagree on that, and i respect that, and you are free to create a PR for this project, but again, I would not want something like that my self to be there as a default.

**Reply by jklaus** [reply]
My thought is during the restoration/deletion process if the account has solely owned orgs it would prompt telling the user the account cannot be deleted until the solely owned orgs are addressed and offer the ability to delete them right then and there.  The user can choose to either delete them right then and there or back out of the process and try to do something different.

**Reply by BlackDex** [reply]
Ok, now i hacked your mail account, and i start the deletion of your account, because i want to. And I just happily delete all the orgs with it. That would be bad not?

In my opinion if you lost your credentials, you lost the rights to delete any organization and you need to contact the administrators/support to help you.
